### ISSUE TITLE
`needless_return` gives confusing advice

### ISSUE THREAD

**Issue by kpreid** [rust-lang/rust-clippy#10816]
### Summary

`needless_return` tells the user to remove the `return` but does not tell them to remove the accompanying semicolon, resulting in a broken program.

### Reproducer

I ran `cargo clippy` on this code:

```rust
fn foo() -> i32 {
    return 100;
}
```

I expected to see this happen: A message that guides me to change the body to `{ 100 }`.

Instead, this happened: A message that guides me to change the body to `{ 100; }`:

```
warning: unneeded `return` statement
 --> src/lib.rs:2:5
  |
2 |     return 100;
  |     ^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
  = note: `#[warn(clippy::needless_return)]` on by default
  = help: remove `return`
```

Note: this is _only_ a problem with the user-visible message. `cargo clippy --fix` _does_ produce a correct program.

### Version

```text
rustc 1.69.0 (84c898d65 2023-04-16)
binary: rustc
commit-hash: 84c898d65adf2f39a5a98507f1fe0ce10a2b8dbc
commit-date: 2023-04-16
host: x86_64-apple-darwin
release: 1.69.0
LLVM version: 15.0.7

rustc 1.71.0-nightly (2f2c438dc 2023-05-08)
binary: rustc
commit-hash: 2f2c438dce75d8cc532c3baa849eeddc0901802c
commit-date: 2023-05-08
host: x86_64-apple-darwin
release: 1.71.0-nightly
LLVM version: 16.0.2
```

### Additional Labels

@rustbot label +I-suggestion-causes-error

**Comment by kpreid** [comment]
Additionally, if using `rust-analyzer` & VS Code, the lint displays _two independently applicable suggestions_, “remove `return`” and “remove this semicolon”. They should be a single suggestion that is only applicable as a unit, in order to produce a correct program.
